qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,379 @@
+        self._absolute = absolute
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """
+        Create the circuit for the spectroscopy experiment. The circuits are based on a
+        GaussianSquare pulse and a frequency_shift instruction encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 1024 samples.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fifth of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+
+        Raises:
+            QiskitError: If relative frequencies are used but no backend was given.
+        """
+        if not backend and not self._absolute:
+            raise QiskitError("Cannot run spectroscopy relative to qubit without a backend.")
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
+        sigma = circuit_options.get("sigma", duration / 4)
+        width = circuit_options.get("width", 0)
+
+        drive = pulse.DriveChannel(self._physical_qubits[0])
+
+        # Create a template circuit
+        freq_param = Parameter("frequency")
+        with pulse.build(name="spectroscopy") as sched:
+            pulse.set_frequency(freq_param, drive)
+            pulse.play(pulse.GaussianSquare(duration, amp, sigma, width), drive)
+
+        gate = Gate(name="Spec", num_qubits=1, params=[freq_param])
+
+        circuit = QuantumCircuit(1)
+        circuit.append(gate, (0,))
+        circuit.add_calibration(gate, (self._physical_qubits[0],), sched, params=[freq_param])
+        circuit.measure_active()
+
+        if not self._absolute:
+            center_freq = backend.defaults().qubit_freq_est[self._physical_qubits[0]]
+
+        # Create the circuits to run
+        circs = []
+        for freq in self._frequencies:
+            if not self._absolute:
+                freq += center_freq
+
+            assigned_circ = circuit.assign_parameters({freq_param: freq}, inplace=False)
+            assigned_circ.metadata = {
+                "experiment_type": self._type,
+                "qubit": self._physical_qubits[0],
+                "xval": freq,
+                "unit": "Hz",
+                "amplitude": amp,
+                "duration": duration,
+                "sigma": sigma,
+                "width": width,
+                "absolute frequencies": self._absolute,
+                "schedule": str(sched),
```

Comment:
Do we need full string representation of schedule? It seems like we need to attach schedule for other complicated experiments, and it may quickly populate the database.

Reply:
It makes it very easy to understand what exactly was run on the backend. I fear that without it it will be hard to understand past results.